File: AutomaticBrewingCoffeeApp/lib/app/data/models/menu.dart
import 'package:abc_androidapp/app/data/models/menu_product_mapping.dart';

class Menu {
  final String menuId;
  final String name;
  final String? description;
  final String status;
  final List<MenuProductMapping> productsInMenu;

  Menu({
    required this.menuId,
    required this.name,
    this.description,
    required this.status,
    required this.productsInMenu,
  });

  factory Menu.fromJson(Map<String, dynamic> json) {
    return Menu(
      menuId: json['menuId'] as String,
      name: json['name'] as String,
      description: json['description'],
      status: json['status'] as String,
      productsInMenu: (json['menuProductMappings'] as List<dynamic>)
          .map((e) => MenuProductMapping.fromJson(e))
          .toList(),
    );
  }
}

File: AutomaticBrewingCoffeeKioskBE/ABC_Kiosk_BE/Domain/Models/Menu.cs
using Domain.Enums;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using System;
using System.ComponentModel.DataAnnotations;

namespace Domain.Models;

public partial class Menu
{
    [Key][StringLength(50)][Required] public string MenuId { get; set; } = null!;

    [StringLength(100)][Required] public string Name { get; set; } = null!;

    [StringLength(300)] public string? Description { get; set; }

    [StringLength(10)][Required] public BaseStatus Status { get; set; } 

    public virtual ICollection<MenuProductMapping> MenuProductMappings { get; set; } = new List<MenuProductMapping>();
}

public class MenuConfig : IEntityTypeConfiguration<Menu>
{
    public void Configure(EntityTypeBuilder<Menu> builder)
    {
        builder
            .Property(o => o.Status)
            .HasConversion(
                    v => v.ToString(),
                    v => (BaseStatus)Enum.Parse(typeof(BaseStatus), v)
                );
    }
}